presto-main/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -68,13 +67,10 @@ public QuerySessionSupplier(
     public Session createSession(QueryId queryId, SessionContext context, WarningCollectorFactory warningCollectorFactory)
     {
         Identity identity = context.getIdentity();
-        accessControl.checkCanSetUser(
```

Comment:
Why is this check removed ? This can let the query run and this is a change in behavior.

Reply:
Have answered in the chat. I moved it mainly for readability: checking permissions is a different logic from creating a session. The behavior change is the session is always created before checking permissions; since we always check permissions after createSession, it should be fine. If we really want to keep the same behavior, we can do the permission check always before the createSession, or call permission check inside it. Which one do you think it is better?